src/Castle.Windsor/MicroKernel/Arguments.cs

```diff
@@ -110,13 +92,17 @@ bool IDictionary.IsReadOnly
 			get { return arguments.IsReadOnly; }
 		}
 
+		public static Arguments Empty { get; } = new Arguments();
```

Comment:
The way I read the code the `Arguments.Empty` collection instance isn't readonly!

Also is there a reason you decided to use a property rather than a field like `String.Empty`, `Guid.Empty`, `Type.EmptyTypes`, `EventArgs.Empty`? Performance-wise it probably isn't a big deal since people aren't going to call it often, but I didn't think the CLR can optimise across assembly boundaries for properties that just return a backing field.

Reply:
Interesting, I always knew that the optimisation you speak of happened for constants. Did not know that was a thing for fields. Does this only happen for fields where the readonly modifier is used? 

Getting rid of the property is a good shout. Less boxing/unboxing.